src/genie_python/genie_simulate_impl.py

```diff
@@ -1225,13 +1246,13 @@ def get_runcontrol_settings(self, name: str) -> tuple():
             self.block_dict[name]["highlimit"],
         )
 
-    def check_alarms(*blocks: str) -> tuple[list[str], list[str], list[str]]:
+    def check_alarms(self, blocks: Tuple[str, ...]) -> tuple[list[str], list[str], list[str]]:
         minor = list()
         major = list()
         invalid = list()
         return (minor, major, invalid)
 
-    def check_limit_violations(self, blocks: str) -> list:
+    def check_limit_violations(self, blocks: typing.Tuple[str, ...]) -> list:
```

Comment:
Pick one of `tuple[str]`, `Tuple[str]` and `typing.Tuple[str]` and use consistently. I'd suggest the first one.